etna/lib/etna/directed_graph.rb

```diff
@@ -18,6 +18,27 @@ def add_connection(parent, child)
     parents[parent] = parent_parents
   end
 
+  def paths_from(root)
```

Comment:
This determines, in an acyclical directed graph, (but with potential convergence and divergence), one of each of the shortest paths to every node that isn't a strict subset of some other longer shortest path.  Which is how we can present a converging and diverging directed graph in a linear stable way.